components/raftstore/src/store/fsm/apply.rs

```diff
@@ -1126,7 +1130,33 @@ where
         }
     }
 
-    fn find_pending(
+    /// Take the pending normal command.
+    /// Note that the `PendingCmd` of this index and term is not popped.
+    fn take_pending_normal_cmd(&mut self, index: u64, term: u64) -> Option<RaftCmdRequest> {
+        while let Some(mut head) = self.pending_cmds.pop_normal(index, term) {
```

Comment:
How about introducing a new method to get a mutable reference instead? So the command doesn't need to be moved twice.